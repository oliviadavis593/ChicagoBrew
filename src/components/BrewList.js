import React, { Component } from 'react';
import BrewItem from '../components/BrewItem';
import Footer from '../views/Footer';
import STORE from '../components/store';
import '../styles/BrewList.css';

class BrewList extends Component {
    render() {
        return(
            <section className='cbt-brew-list__container'>
                <ul className='cbt-brew_list'>
                    {/*Map through the store file to get the entire list of breweries */}
                  {STORE.brews.map(brew => (
                      <li key={brew.id}>
                          <BrewItem
                          id={brew.id}
                          name={brew.name}
                          />
                      </li>
                  ))}
                </ul>
                <div className='footer'>
                    <Footer />
                </div>
            </section>
        )
    }
}

export default BrewList; 