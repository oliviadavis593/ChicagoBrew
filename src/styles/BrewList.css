.cbt-brew_list {
    text-align: center; 
    display: inline; 
}

.cbt-brew_list li {
    list-style: none; 
}

.cbt-brew_list a {
    text-decoration: none; 
    color: #000; 
}