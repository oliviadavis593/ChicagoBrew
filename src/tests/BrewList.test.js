import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import BrewList from '../components/BrewList';


describe('BrewList component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrewList />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer 
            .create(<BrewList />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})