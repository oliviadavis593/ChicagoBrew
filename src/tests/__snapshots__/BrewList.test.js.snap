// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrewList component renders the UI as expected 1`] = `
<section
  className="cbt-brew-list__container"
>
  <ul
    className="cbt-brew_list"
  />
  <div
    className="footer"
  >
    <footer
      className="footer"
    >
      <div
        className="footer__brand-container"
      >
        <span
          role="img"
        >
          Made with 
          <span
            aria-label="heart-emoji"
            role="img"
          >
            ❤️
          </span>
           by 
          <a
            href="https://oliviadavis593.github.io/portfolio/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Olivia Davis
          </a>
        </span>
      </div>
      <div
        className="copyright-text"
      >
        <p>
          Copyright © 2020
          <br />
          All Rights Reserved
        </p>
      </div>
    </footer>
  </div>
</section>
`;
